providers/implementations/rands/drbg.c

```diff
@@ -0,0 +1,1043 @@
+    }
+
+    if (drbg->reseed_interval > 0) {
+        if (drbg->reseed_gen_counter > drbg->reseed_interval)
+            reseed_required = 1;
+    }
+    if (drbg->reseed_time_interval > 0) {
+        time_t now = time(NULL);
+        if (now < drbg->reseed_time
+            || now - drbg->reseed_time >= drbg->reseed_time_interval)
+            reseed_required = 1;
+    }
+    if (drbg->parent != NULL
+            && get_parent_reseed_count(drbg) != drbg->parent_reseed_counter)
+        reseed_required = 1;
+
+    if (reseed_required || prediction_resistance) {
+        if (!PROV_DRBG_reseed(drbg, prediction_resistance, NULL, 0,
+                              adin, adinlen)) {
+            PROVerr(0, PROV_R_RESEED_ERROR);
+            return 0;
+        }
+        adin = NULL;
+        adinlen = 0;
+    }
+
+    if (!drbg->generate(drbg, out, outlen, adin, adinlen)) {
+        drbg->state = EVP_RAND_STATE_ERROR;
+        PROVerr(0, PROV_R_GENERATE_ERROR);
+        return 0;
+    }
+
+    drbg->reseed_gen_counter++;
+
+    return 1;
+}
+
+/*
+ * Restart |drbg|, using the specified entropy or additional input
+ *
+ * Tries its best to get the drbg instantiated by all means,
+ * regardless of its current state.
+ *
+ * Optionally, a |buffer| of |len| random bytes can be passed,
+ * which is assumed to contain at least |entropy| bits of entropy.
+ *
+ * If |entropy| > 0, the buffer content is used as entropy input.
+ *
+ * If |entropy| == 0, the buffer content is used as additional input
+ *
+ * Returns 1 on success, 0 on failure.
+ *
+ * This function is used internally only.
+ */
+static int rand_drbg_restart(PROV_DRBG *drbg)
+{
+    if (drbg->seed_pool != NULL) {
+        drbg->state = EVP_RAND_STATE_ERROR;
+        rand_pool_free(drbg->seed_pool);
+        drbg->seed_pool = NULL;
+        RANDerr(0, ERR_R_INTERNAL_ERROR);
+        return 0;
+    }
+
+    /* repair error state */
+    if (drbg->state == EVP_RAND_STATE_ERROR)
+        drbg->uninstantiate(drbg);
+
+    /* repair uninitialized state */
+    if (drbg->state == EVP_RAND_STATE_UNINITIALISED)
+        /* reinstantiate drbg */
+        PROV_DRBG_instantiate(drbg, drbg->strength, 0, NULL, 0);
+
+    rand_pool_free(drbg->seed_pool);
+    drbg->seed_pool = NULL;
+    return drbg->state == EVP_RAND_STATE_READY;
+}
+
+/* Provider support from here down */
+static const OSSL_DISPATCH *find_call(const OSSL_DISPATCH *dispatch,
+                                      int function)
+{
+    if (dispatch != NULL)
+        while (dispatch->function_id != 0) {
+            if (dispatch->function_id == function)
+                return dispatch;
+            dispatch++;
+        }
+    return NULL;
+}
+
+int drbg_enable_locking(void *vctx)
+{
+    PROV_DRBG *drbg = vctx;
+
+    if (drbg != NULL && drbg->lock == NULL) {
+    /*
+        if (drbg->state != EVP_RAND_STATE_UNINITIALISED) {
+            ERR_raise(ERR_LIB_PROV, PROV_R_DRBG_ALREADY_INITIALIZED);
+            return 0;
+        }
+        */
```

Comment:
Why commented out code?

Reply:
It was left in in case I needed to put it back.  I don't so have removed it.